src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -278,4 +278,15 @@ public final void log(int lineNo, int colNo, String key,
                 getClass(),
                 getCustomMessages().get(key)));
     }
+
+    /**
+     * The actual context holder.
+     */
+    private static class Context {
```

Comment:
`> Also since it contains contents of a file, file violations and file contents, shouldn't it be named something like FileContextHolder?

Renamed to `FileContext` - I'm not sure if `Holder` adds more information here